src/main/java/org/concordion/internal/XMLSpecification.java

```diff
@@ -46,7 +45,7 @@ public XMLSpecification(CommandCall rootCommandNode) {
 
     public void processNode(CommandCall node, Evaluator evaluator, ResultRecorder resultRecorder) {
 
-        if (hasNonExampleChildren(node)) {
+        if (node.hasNonExampleChildren()) {
```

Comment:
as per comment on CommandCall#hasNonExampleChildren, this would be something more like:
`if (node.hasNonExampleChildren() || node.executesAsExample())`